test/types/maps.test.ts

```diff
@@ -71,3 +71,19 @@ function gh10872(): void {
 
   doc.toJSON().map1.foo;
 }
+
+function gh13755() {
+  class Test {
+    instance: Map<string, string>;
+    constructor() {
+      this.instance = new Map<string, string>();
+    }
+  }
+
+  const testSchema = new Schema<Test>({
```

Comment:
i think this test is not quite obvious what it is testing and does not actually test what this fixes (add a expectType?), a different example:

(mongoose 7.6.1)

```ts
const sch = new mongoose.Schema({
  prop: {
    type: 'Map',
    of: 'String',
  },
  prop2: {
    type: 'String',
  },
});

const model = mongoose.model('Test', sch);

const doc = new model();
doc.prop; // Type: unknown
doc.prop2; // Type: string | undefined
```

removing the second property, the type of `prop` becomes:
```txt
prop: {
    type: "Map";
    of: "String";
}
```

and adding the class to this, would only result in a error that the `prop` types are incompatible:
```txt
Type '{ type: string; of: string; }' is not assignable to type 'SchemaDefinitionProperty<Map<string, string>, Test> | undefined'.
  Types of property 'type' are incompatible.
    Type 'string' is not assignable to type 'typeof Mixed | { readonly prototype?: SchemaDefinitionProperty<Map<any, any>, any> | undefined; readonly [Symbol.species]?: SchemaDefinitionProperty<MapConstructor, any> | undefined; } | undefined'.ts(2322)
```

Reply:
Good call, fixed in https://github.com/Automattic/mongoose/pull/13960/commits/c4416a60ce596be64829619e48f8fabadfdab0f6